tests/test_v2x.py

```diff
@@ -0,0 +1,176 @@
+        relhref = xmlinc.make_relhref(left, right)
+        leftcontent = leftfile.read()
+        if relhref in leftcontent:
+            return 1
+        leftinccount = leftcontent.count('`include')
+    with open(right, 'r') as rightfile:
+        relhref = xmlinc.make_relhref(right, left)
+        rightcontent = rightfile.read()
+        if relhref in rightcontent:
+            return -1
+        rightinccount = rightcontent.count('`include')
+    return leftinccount - rightinccount
+
+
+def find_files(pattern, rootdir):
+    return [str(f) for f in Path(os.path.abspath(rootdir)).glob(pattern)]
+
+
+def get_test_goldens(goldentype, testfile):
+    simfiles = sorted(
+        sorted(convert.get_filenames_containing('*.sim.v', testfile)),
+        key=cmp_to_key(order_based_on_deps))
+    goldens = []
+    for sim in simfiles:
+        res = find_files('golden.' + goldentype, os.path.dirname(sim))
+        if len(res) == 1:
+            goldens.append({
+                'simfile': sim,
+                'goldenfile': res[0]})
+    return goldens
+
+
+def pytest_generate_tests(metafunc):
+    prepare_files()
+    scriptdir = os.path.dirname(__file__)
+    testdir = os.path.join(os.path.dirname(scriptdir), TEST_TMP_SUFFIX)
+
+    if "modelcase" in metafunc.fixturenames:
+        models = get_test_goldens('model.xml', os.path.join(testdir, 'test'))
+        metafunc.parametrize("modelcase",
+                             models,
+                             ids=[i['simfile'] for i in models])
+    if "pbtypecase" in metafunc.fixturenames:
+        models = get_test_goldens('pb_type.xml', os.path.join(testdir, 'test'))
+        metafunc.parametrize("pbtypecase",
+                             models,
+                             ids=[i['simfile'] for i in models])
+
+
+def test_model_generation_with_vlog_to_model(modelcase):
+    """Parametrized test that checks  if the model.xml files produced by the
+    vlog_to_model function are valid
+    Parameters
+    ----------
+    modelcase : dict
+        A dict of the filename of the model.xml file that should be produced
+        by the corresponding sim.v file ('goldenfile') and the corresponding
+        sim.v file ('simfile')
+    """
+    modelfile = modelcase['goldenfile']
+    testdatadir = os.path.dirname(modelfile) + '/'
+    vlog_filename = modelcase['simfile']
+    testname = vlog_filename.split('/')[-1].split('.')[0]
+    generatedmodelfile = os.path.join(testdatadir, testname) + '.model.xml'
+    modelout = vlog_to_model.vlog_to_model([vlog_filename], None, None,
+                                           generatedmodelfile)
+    with open(generatedmodelfile, 'w') as model:
+        model.write(modelout)
+
+    convertedgolden = convert.vtr_stylize_xml(modelfile)
+    convertedmodel = convert.vtr_stylize_xml(generatedmodelfile)
+
+    with open(generatedmodelfile + '.actual', 'w') as model:
+        model.write(convertedmodel)
+
+    assert convertedmodel == convertedgolden
```

Comment:
During the test collection process, the pytest module scans the collected functions for asserts and rewrites them so they print the information about the assertion error, including diffs for multiline strings. It is called assertion introspection, and is mentioned in http://doc.pytest.org/en/latest/assert.html, and described here http://doc.pytest.org/en/latest/assert.html#assertion-introspection-details.

This enables using simple operators in the pytest tests, instead of asserts from `unittest`, without loosing precise error information.